features/step_definitions/logging.rb

```diff
@@ -301,12 +301,27 @@
   step %Q/I wait for the resource "daemonset" named "fluentd" to disappear/
 end
 
-Given /^I run curl command on the CLO pod to get metrics with:$/ do | table |
+Given /^I run curl command on the (CLO|ES|fluentd) pod to get metrics with:$/ do | pod_type, table |
   ensure_admin_tagged
   opts = opts_array_to_hash(table.raw)
-  step %Q/a pod becomes ready with labels:/, table(%{
+  case pod_type
+  when "CLO"
+    step %Q/a pod becomes ready with labels:/, table(%{
       | name=cluster-logging-operator |
     })
+    container_name = "cluster-logging-operator"
+  when "ES"
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | cluster-name=elasticsearch,component=elasticsearch |
+    })
+    container_name = "elasticsearch"
+  when "fluentd"
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | logging-infra=fluentd |
+    })
+    container_name = "fluentd"
+  end
```

Comment:
Maybe make it more readable like below ?
```
  case pod_type
  when "CLO"
    selector = "name=cluster-logging-operator"
    container_name = "cluster-logging-operator"
  when "ES"
    selector = "cluster-name=elasticsearch,component=elasticsearch"
    container_name = "elasticsearch"
  when "fluentd"
    selector = "logging-infra=fluentd"
    container_name = "fluentd"
  end
  step %Q/a pod becomes ready with labels:/, table(%{
    | #{selector} |
  })
```

Reply:
@liangxia Good idea, thanks a lot.